Source/WebKit/UIProcess/Cocoa/SystemPreviewControllerCocoa.mm

```diff
@@ -281,15 +280,18 @@ - (UIImage *)previewController:(QLPreviewController *)controller transitionImage
 
     [presentingViewController presentViewController:m_qlPreviewController.get() animated:YES completion:nullptr];
 #endif
+
+    m_originatingPageURL = originatingPageURL;
 }
 
 void SystemPreviewController::setDestinationURL(URL url)
 {
 #if HAVE(UIKIT_WEBKIT_INTERNALS)
     url.removeFragmentIdentifier();
     NSURL *nsurl = (NSURL *)url;
-    if ([getASVLaunchPreviewClass() respondsToSelector:@selector(beginPreviewApplicationWithURLs:is3DContent:completion:)])
```

Comment:
Do you want to keep the old path around?

Reply:
It's still there so I'll add an `else if`